### PR TITLE
Replace the deprecated json-c functions

### PR DESCRIPTION
**Pull request by belltoy**

### CODE REVIEW

File: nsqlookupd.c [hunk 1]
```diff
@@ -34,15 +34,14 @@ void nsq_lookupd_request_cb(struct HttpRequest *req, struct HttpResponse *resp,
         return;
     }
 
-    data = json_object_object_get(jsobj, "data");
-    if (!jsobj) {
```
Comment (belltoy) [inline nsqlookupd.c]: Here should be `data`.

**Comment by mreiferson** [comment]
@belltoy looks good, do you mind squashing down to one commit?

**Comment by belltoy** [comment]
@mreiferson Sure

**Comment by mreiferson** [comment]
LGTM, thanks!

**Pull request merged**